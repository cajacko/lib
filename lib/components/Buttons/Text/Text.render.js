// @flow

import React from 'react';
import Button from '../Button';
import Text from '../../Text';

const ButtonsText = ({ action, text, ...props }) => (
  <Button action={action} {...props}>
    <Text text={text} {...props} />
  </Button>
);

export default ButtonsText;
