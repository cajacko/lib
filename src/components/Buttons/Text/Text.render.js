import React from "react";

const ButtonsText = ({ text }) => <button>{text}</button>;

export default ButtonsText;
