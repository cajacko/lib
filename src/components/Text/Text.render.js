import React from "react";

/**
 * Render a simple text element
 *
 * @param {Object} props Props passed to the component
 * @param {String} props.text The text to render
 *
 * @return {ReactNode} Markup to render
 */
const Text = ({ text }) => <span>{text}</span>;

export default Text;
