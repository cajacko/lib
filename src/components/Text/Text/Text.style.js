import styled from "styled-components";
import { FONT_FAMILY, FONT_SIZES, FONT_COLORS } from "../../../config/fonts";

const getColour = ({ color }) => {
  return FONT_COLORS[color] || FONT_COLORS.DEFAULT;
};

export const Span = styled.span`
  font-family: ${FONT_FAMILY.DEFAULT};
  font-size: ${FONT_SIZES.DEFAULT}px;
  color: ${getColour};
`;
