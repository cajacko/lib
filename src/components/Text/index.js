export { default } from "./Text.render";
