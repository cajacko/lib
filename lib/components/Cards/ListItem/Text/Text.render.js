import React from 'react';
import Text from '../../../Text';

const CardsListItemText = ({ text }) => <Text text={text} type="body2" />;

export default CardsListItemText;
