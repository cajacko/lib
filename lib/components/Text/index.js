// @flow

import Text from './Text.render';
import withTextTransform from '../HOCs/withTextTransform';

export default withTextTransform(Text);
