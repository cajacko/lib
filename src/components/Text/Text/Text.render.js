import React from "react";
import { Span } from "./Text.style";

const TextText = ({ text, color }) => <Span color={color}>{text}</Span>;

export default TextText;
