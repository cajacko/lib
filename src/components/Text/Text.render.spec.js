import React from "react";
import Renderer from "react-test-renderer";
import Text from "./Text.render";

describe("Test.render", () => {
  it("Renders the passed text", () => {
    const component = Renderer.create(<Text text="Hello" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
