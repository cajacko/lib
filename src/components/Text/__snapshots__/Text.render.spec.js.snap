// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test.render Renders the passed text 1`] = `
<span>
  Hello
</span>
`;
